api/cmd/cache_verify.go

```diff
@@ -0,0 +1,177 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cmd
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/spf13/cobra"
+	"github.com/tidwall/gjson"
+
+	"github.com/apisix/manager-api/internal/conf"
+	"github.com/apisix/manager-api/internal/handler/cache_verify"
+	"github.com/apisix/manager-api/internal/log"
+)
+
+var port int
+var host string
+
+var username, password string
+
+type response struct {
+	Data cache_verify.OutputResult `json:"data"`
+}
+
+func newCacheVerifyCommand() *cobra.Command {
+	return &cobra.Command{
+		Use:   "cache-verify",
+		Short: "verify that data in cache are consistent with that in ETCD",
+		Run: func(cmd *cobra.Command, args []string) {
+			conf.InitConf()
+			log.InitLogger()
+
+			port = conf.ServerPort
+			host = conf.ServerHost
+			username = "admin"
+			password = conf.UserList[username].Password
+
+			token := getToken()
+
+			url := fmt.Sprintf("http://%s:%d/apisix/admin/cache_verify", host, port)
+			client := &http.Client{}
+
+			get, err := http.NewRequest("GET", url, nil)
+			if err != nil {
+				log.Errorf("new http request failed: %s", err)
+				return
+			}
+
+			get.Header.Set("Authorization", token)
+
+			rsp, err := client.Do(get)
+			if err != nil {
+				log.Errorf("get result from migrate/export failed: %s", err)
+				return
+			}
+			defer func() {
+				err := rsp.Body.Close()
+				if err != nil {
+					log.Errorf("close on response body failed: %s", err)
+					return
+				}
+			}()
+
+			data, err := ioutil.ReadAll(rsp.Body)
+			if err != nil {
+				log.Errorf("io read all failed: %s", err)
+				return
+			}
+
+			var rs response
+			err = json.Unmarshal(data, &rs)
+			if err != nil {
+				log.Errorf("bad Data format,json unmarshal failed: %s", err)
+				return
+			}
+
+			fmt.Printf("cache verification result as follows:\n\n")
+			fmt.Printf("There are %d items in total,%d of them are consistent,%d of them are inconsistent\n",
+				rs.Data.Total, rs.Data.ConsistentCount, rs.Data.InconsistentCount)
+
+			names := []string{
+				"ssls",
+				"routes",
+				"scripts",
+				"services",
+				"upstreams",
+				"consumers",
+				"server infos",
+				"global plugins",
+				"plugin configs",
+			}
+			datas := []cache_verify.StatisticalData{
+				rs.Data.Items.SSLs,
+				rs.Data.Items.Routes,
+				rs.Data.Items.Scripts,
+				rs.Data.Items.Services,
+				rs.Data.Items.Upstreams,
+				rs.Data.Items.Consumers,
+				rs.Data.Items.ServerInfos,
+				rs.Data.Items.GlobalPlugins,
+				rs.Data.Items.PluginConfigs,
+			}
```

Comment:
why not use a map? and we could put them into the function `printResult`

Reply:
because map doesn't guarantee iteration order,I want to make sure the output are sorted by the config name,like:
```
❯ go run main.go cache-verify 
cache verification result as follows:

There are 12 items in total,12 of them are consistent,0 of them are inconsistent
ssls           : 0 in total,0 consistent,0 inconsistent
routes         : 1 in total,1 consistent,0 inconsistent
scripts        : 0 in total,0 consistent,0 inconsistent
services       : 1 in total,1 consistent,0 inconsistent
upstreams      : 2 in total,2 consistent,0 inconsistent
consumers      : 7 in total,7 consistent,0 inconsistent
server infos   : 1 in total,1 consistent,0 inconsistent
global plugins : 0 in total,0 consistent,0 inconsistent
plugin configs : 0 in total,0 consistent,0 inconsistent
```